src/web/content.rs

```diff
@@ -0,0 +1,179 @@
+use std::io::Read;
+
+use response::Responder;
+use serde::de::Deserialize;
+use serde::Serialize;
+
+use rocket::data::{Data, FromData, Outcome, Transform, Transform::*, Transformed};
+use rocket::http::{Accept, ContentType, QMediaType, Status};
+use rocket::outcome::Outcome::*;
+use rocket::request::{self, FromRequest, Request};
+use rocket::response;
+use rocket_contrib::json::Json;
+
+use crate::errors::ApiError;
+
+/// The default limit for the request size (to prevent DoS attacks). Can be
+/// overriden in the config by setting `max_request_size` to a different value
+const MAX_REQUEST_SIZE: u64 = 1024 * 1024;
+/// The list of supported formats. When changed, the implementation of
+/// Input::from_data() must be updated accordingly.
+const SUPPORTED_MEDIA_TYPES: [ContentType; 1] = [ContentType::JSON];
+const PREFERRED_MEDIA_TYPE: ContentType = ContentType::JSON;
+
+fn supported_media_type_message() -> String {
```

Comment:
Yeah, perhaps this code is trying too hard to be clever to avoid a bit of duplication. Given that  we are only going to support a handful of content types, maybe it's easier to just hard code a static message indeed.

Reply:
Made the message static (note, that it's also used in some `ApiError` variants).